mobile/src/navigation/app/app.navigation.tsx

```diff
@@ -22,15 +16,18 @@ const App: FC = () => {
       screenOptions={SCREEN_OPTIONS}
       drawerContent={(props): JSX.Element => <DrawerContent {...props} />}
     >
-      <Drawer.Screen name={AppScreenName.OVERVIEW} component={Overview} />
-      <Drawer.Screen name={AppScreenName.COURSES} component={Courses} />
-      <Drawer.Screen name={AppScreenName.MENTORS} component={Mentors} />
-      <Drawer.Screen
-        name={AppScreenName.MY_EDUCATION}
-        component={MyEducation}
-      />
-      <Drawer.Screen name={AppScreenName.BILLING} component={Billing} />
-      <Drawer.Screen name={AppScreenName.SETTINGS} component={Settings} />
+      {NAVIGATION_ITEMS.map((item) => {
+        return item.subroutes.map((subItem) => {
+          if (subItem.permissions.includes('uac')) {
```

Comment:
Do it via useMemo and create a helper for checking permissions. Will get user permissions when we fetch user. It should look like this

```
const { permissions } = useAppSelector((state) => state.auth.user);

const allowedScreens = useMemo(() => {
  const screens = item.map(subroutes => subroutes);
 
  return getAllowedScreens(screens, permissions);
}, [permissions]);
```